model/ServerHandler.php

```diff
@@ -0,0 +1,167 @@
+            $this->msg['store_error'] = 'pAdminEdit_server_result_error';
+            $this->msg['successmessage'] = 'server_updated';
+        }
+        $this->msg['can_create'] = $this->is_superadmin;
+    }
+
+    public function webformConfig() {
+        return array(
+            # $PALANG labels
+            'formtitle_create' => 'pAdminCreate_server_welcome',
+            'formtitle_edit' => 'pAdminEdit_server_welcome',
+            'create_button' => 'pAdminCreate_server_button',
+
+            # various settings
+            'required_role' => 'admin',
+            'listview' => 'list.php?table=server',
+            'early_init' => 0,
+        );
+    }
+
+    protected function storemore() {
+        if ($this->new) {
+            if (!$this->server_postcreation()) {
+                $this->errormsg[] = Config::lang('server_postcreate_failed');
+            }
+        }
+        return true;
+    }
+
+    protected function server_postcreation() {
+        $script=Config::read('server_postcreation_script');
+
+        if (empty($script)) {
+            return true;
+        }   
+
+        if (empty($this->id)) {
+            $this->errormsg[] = 'Empty server parameter in server_postcreation';
+            return false;
+        }   
+
+        $cmdarg1=escapeshellarg($this->id);
+        $command= "$script $cmdarg1";
+        $retval=0;
+        $output=array();
+        $firstline='';
+        $firstline=exec($command,$output,$retval);
+        if (0!=$retval) {
+            error_log("Running $command yielded return value=$retval, first line of output=$firstline");
+            $this->errormsg[] = 'Problems running server postcreation script!';
+            return FALSE;
+        }   
+
+        return TRUE;
+
+    }
+
+    protected function server_postdeletion() {
+        $script=Config::read('server_postdeletion_script');
+
+        if(empty($script)) {
+            return true;
+        }
+
+        if (empty($this->id)) {
+        }
+
+        $cmdarg1=escapeshellarg($this->id);
+        $command= "$script $cmdarg1";
+        $retval=0;
+        $output=array();
+        $firstline='';
+        $firstline=exec($command,$output,$retval);
+        if (0!=$retval) {
+            error_log("Running $command yielded return value=$retval, first line of output=$firstline");
+            $this->errormsg[] = 'Problems running server postdeletion scripts!';
+            return false;
+        }
+
+        return true;
+    }
+
+    public function delete() {
+        if (! $this->is_superadmin) {
+            $this->errormsg[] = Config::Lang_f('no_delete_permissions', $this->id);
+            return false;
+        }
+        
+        if ( ! $this->view() ) {
+            $this->errormsg[] = Config::Lang('server_does_not_exist');
+            return false;
+        }
+
+        $handler = new DomainHandler(0, $this->admin_username);
+        $handler->getList("primarymx = '" . escape_string($this->id) . "'");
```

Comment:
You can use an array as getList parameter:

    $handler->getList( array("primarymx" => $this->id) );

Note that this will enable some additional checks in getList, but that (hopefully) shouln't hurt.